app/knight.py

```diff
@@ -51,13 +40,7 @@ def use_potion(self) -> None:
             elif key == "protection":
                 self.protection += value
 
-    def __str__(self) -> str:
-        return (
-            f"Knight: {self.name} "
-            f"Power: {self.power} "
-            f"HP: {self.hp} "
-            f"Armour: {self.armour} "
-            f"Weapon: {self.weapon.__dict__} "
-            f"Potion: {self.potion} "
-            f"Protection: {self.protection}"
-        )
+    def duel_battle(self: Knight, other: Knight) -> None:
+        self.hp = self.hp - (other.get_power() - self.get_protection())
+        if self.hp < 0:
+            self.hp = 0
```

Comment:
if hp == 0, the knight is dead

Reply:
So i have to implement method is_dead() for knight? That would set hp to 0